src/librustc_mir/monomorphize/polymorphize.rs

```diff
@@ -245,6 +245,13 @@ impl<'a, 'tcx> TypeVisitor<'tcx> for UsedGenericParametersVisitor<'a, 'tcx> {
                 self.unused_parameters.clear(param.index);
                 false
             }
+            ty::ConstKind::Unevaluated(_, _, Some(p)) => {
```

Comment:
We actually have the same problem for other unevaluated constants:
```rust
#![feature(lazy_normalization_consts)]

fn test<T>() {
    let x = [0; 3 + 4];
}

pub fn caller() {
    test::<String>();
    test::<Vec<String>>();
}
```
But afaict that doesn't concern stable rn, not completely sure if we can end up with cycle errors here, but I think we should be fine and able to use `unused_generic_params` here (or we just have to restrict this to `DefKind::AnonConst` for now)

Reply:
Fixed!